encoding/encoding.go

```diff
@@ -36,17 +42,30 @@ func Encode(value Encodable) ([]byte, error) {
 	return cbor.DumpObject(value)
 }
 
+func EncodeToNode(encodable Encodable) (datamodel.Node, error) {
+	byts, err := Encode(encodable)
+	if err != nil {
+		return nil, err
+	}
+	na := basicnode.Prototype.Any.NewBuilder()
+	if err := dagcbor.Decode(na, bytes.NewReader(byts)); err != nil {
+		return nil, err
+	}
+	return na.Build(), nil
+}
+
 // Decoder is CBOR decoder for a given encodable type
 type Decoder interface {
 	DecodeFromCbor([]byte) (Encodable, error)
+	DecodeFromNode(datamodel.Node) (Encodable, error)
 }
 
 // NewDecoder creates a new Decoder that will decode into new instances of the given
 // object type. It will use the decoding that is optimal for that type
 // It returns error if it's not possible to setup a decoder for this type
 func NewDecoder(decodeType Encodable) (Decoder, error) {
-	// check if type is ipld.Node, if so, just use style
-	if ipldDecodable, ok := decodeType.(ipld.Node); ok {
+	// check if type is datamodel.Node, if so, just use style
+	if ipldDecodable, ok := decodeType.(datamodel.Node); ok {
 		return &ipldDecoder{ipldDecodable.Prototype()}, nil
```

Comment:
should we be checking for schema.TypedPrototype here and using Prototype().Representation()

Reply:
probably doesn't affect this PR but I expect it affects the next one?

Reply:
good Q, but maybe it doesn't even matter once we're dealing with a `datamodel.Node` interface, I think this whole encoding package can go away.